helix-core/src/main/java/org/apache/helix/controller/rebalancer/waged/WagedRebalancer.java

```diff
@@ -216,6 +217,11 @@ public void close() {
               itemKeys.addAll(getChangeDetector().getRemovalsByType(changeType));
               return itemKeys;
             }));
+    // Filter for the items that have content changed.
+    clusterChanges =
```

Comment:
cannot you filter in line 211 or inside change detector? Another round of for loop of everything seems wasted

Reply:
Sounds good, I will do it.

Reply:
I tried it. But the stream method becomes too complicated. It hurts the readability. I would keep this logic separately for easier reading. The additional cost is low, considering the number of types is limited.